addons/website_forum/controllers/main.py

```diff
@@ -164,23 +172,40 @@ def tag_read(self, q='', l=25, **post):
         )
         return simplejson.dumps(data)
 
-    @http.route(['/forum/<model("forum.forum"):forum>/tag'], type='http', auth="public", website=True)
-    def tags(self, forum, page=1, **post):
-        tags = request.env['forum.tag'].search([('forum_id', '=', forum.id), ('posts_count', '>', 0)], limit=None, order='posts_count DESC')
+    @http.route(['/forum/<model("forum.forum"):forum>/tag','/forum/<model("forum.forum"):forum>/tag/<tag_char>'], type='http', auth="public", website=True)
+    def tags(self, forum, page=1, tag_char=False, **post):
+        pager_chars = forum.get_tags_char()
+        tag_char = urllib.unquote(tag_char).decode("unicode_escape") if tag_char else (pager_chars[0][0] if pager_chars else 'All')
+        if pager_chars: pager_chars.insert(0,('All',0))
+        domain = [('forum_id', '=', forum.id), ('posts_count', '>', 0)]
+        if tag_char.lower() != "all":
+            domain.append(('name','=ilike',tag_char+'%'))
+        tags = request.env['forum.tag'].search(domain, limit=None, order='posts_count DESC' if tag_char=='All' else 'name')
         values = self._prepare_forum_values(forum=forum, searches={'tags': True}, **post)
         values.update({
             'tags': tags,
             'main_object': forum,
+            'pager_chars': pager_chars,
+            'active_tag':tag_char,
```

Comment:
transform it into an upper letter, otherwise, if you compare the active_char with the keys of pager_chars, it can not match, if a lowercase is introduce in the url.